# Optimización de maquinaria para producción de varillas

<img style="float: right; margin: 0px 0px 15px 15px;" src="https://upload.wikimedia.org/wikipedia/commons/d/db/Logo_ITESO_normal.jpg" width="130px" height="80px" />

> # Integrantes

>- Samir Navarrete
>- Fernando Padilla
>- Oscar Villa

# Objetivos

+ El objetivo de este trabajo, consta de optimizar la produccion de varillas en una empresa para poder maximizar las ganancias por unidad de varilla.

## Objetivo general

+ Encontrar un modelo que nos pueda maximizar las ganancias de la empresa.

## Objetivos especificos

+ Obtener las cantidades óptimas de producción mediante la programación lineal.

+ Mostrar gráficamente el resultado obtenido.

+ Analizar la situación en cuanto a costo-beneficio de la producción de varillas.


+ Los componenetes a tomar en cuenta son:

     >+ Dias en los que se opera 
     >+ Produccion por maquinaria
     >+ Tipos de varilla
     >+ Beneficios 

Tomando en cuenta estos componenetes podremos dar un modelo que garantice la optimizacion de las ganancias, para brindar a la empresa una opcion de produccion en la cual aprovechen mas el uso de maquinaria y aporten un beneficio economico al mismo.
La empresa cuenta con 3 maquinarias para la produccion de varillas.


+ En la maquinaria A se invierten:
   + 4 D-O para producir 1 monto(1000 kg) de varilla de 1"
   + 3 D-O para producir 1 monto de varilla de 3/4"
   + 2 D-O para producir 1 monto de varilla de 1/2"

+ En la maquinaria B se invierten: 
    + 3 D-O para producir 1 monto de varilla de 1"
    + 3 D-O para producir 1 monto de varilla de 3/4"
    + 2 D-O para producir 1 monto de varilla de 1/2"
  

+ En la maquinaria C se invierten:
    + 3 D-O para producir 1 monto de varilla de 1"
    + 3 D-O para producir 1 monto de varilla de 3/4"
    + 2 D-O para producir 1 monto de varilla de 1/2"
 

Si los beneficios son:
+ 1" = 25,000
+ 3/4" = 22,000
+ 1/2" = 18,000



+ **Como se quiere maximizar las ganancias, entonces:**

$$MaxG_{A,B,C}= 25(x_{1A}+x_{1B}+x_{1C})+22(x_{2A}+ x_{2B}+x_{2C})+18(x_{3A}+ x_{3B}+x_{3C})$$

+ **Donde X (V,M), son:**

    + Tipo de Varilla  V(1,2,3)
    
    + Maquinaria  M(A,B,C)


+ **Además, sujeta:**

    - $ 4x_{1A}+ 3x_{2A}+ 3x_{3A}\leq 350$,
    - $ 3x_{1B}+ 3x_{2B}+ 2x_{3B}\leq 320$,
    - $ 3x_{1C}+ 3x_{2C}+ 2x_{3C}\leq 300$,
    - $ x_{1A}, x_{1B}, x_{1C}, x_{2A}, x_{2B}, x_{2C}, x_{3A}, x_{3B}, x_{3C}\geq 0$.

In [8]:
import sympy as sym
import numpy as np
import scipy.optimize as opt

In [9]:
c = np.array([1, 1, 1])
A = -np.array([[4, 3, 3],
               [3, 3, 2],
               [3, 3, 2]])
b = -np.array([350, 320, 300])
x1_cota = (0,None)
x2_cota = (0,None)
x3_cota = (0,None)

res_ganancia = opt.linprog(c, A_ub=A, b_ub=b, bounds = (x1_cota, x2_cota, x3_cota))
res_ganancia

     fun: 106.66666666666667
 message: 'Optimization terminated successfully.'
     nit: 3
   slack: array([  0.,   0.,  20.])
  status: 0
 success: True
       x: array([ 30.        ,  76.66666667,   0.        ])

Debemos producir 30 montos en la máquina A y 76.6667 montos en la máquina B. Sin embargo, el resultado nos arroja que la máquina C está de más.

Con las limitantes que se proveen en el problema, se llega a la conclusión que se deben producir 106.6667 montos de varilla, se debe omitir la maquinaria C para maximizar las ganancias, dado que la relación costo beneficio no es óptima.
